### PR TITLE
Add CPU Registers

### PR DESCRIPTION
**Pull request by Chris-Johnston**
Work on getting the functionality of the CPU registers added to the M68k class

### CODE REVIEW

File: src/easier68k/simulator/m68k.py [hunk 1]
```diff
@@ -11,25 +13,59 @@ def __init__(self):
...
+
+        # loop through all of the 'normal' registers which are just 32 bits / 4 bytes long
+        normal_registers = [Register.D0, Register.D1, Register.D2, Register.D3, Register.D4, Register.D5, Register.D6,
+                            Register.D6, Register.D7, Register.A0, Register.A1, Register.A2, Register.A3, Register.A4,
+                            Register.A5, Register.A6, Register.A7, Register.PC ]
+        for register in normal_registers:
+            self.registers[register] = bytearray(4)
+
+        # set up all of the odd registers (in this case, just the Condition Code Register)
+        # which just uses 5 bits out of the lowermost byte (do we want to allocate it an entire word instead?)
```
Comment (AdamuKaapan) [inline src/easier68k/simulator/m68k.py]: Doesn't the CCR use a whole register and the rest of the status register is used for other stuff? Do we want to implement that or are we storing them as separate variables?

Comment (Chris-Johnston) [reply src/easier68k/simulator/m68k.py]: I'm going off of page 13 of the manual

**Comment by Chris-Johnston** [comment]
Still WIP

File: src/easier68k/simulator/m68k.py (hunk 1 above)
Comment (AdamuKaapan) [reply src/easier68k/simulator/m68k.py]: Ah, I was going off of what I saw in sim68k

File: src/easier68k/core/enum/register.py [hunk 2]
```diff
@@ -13,11 +14,29 @@ class Register:
...
+
+    # constants that group together the registers
+    DATA_REGISTERS = [D0, D1, D2, D3, D4, D5, D6, D7]
+    ADDRESS_REGISTERS = [A0, A1, A2, A3, A4, A5, A6, A7]
+
+    # all of the address registers that are limited to fit within the bounds of memory 2^24
+    MEMORY_LIMITED_ADDRESS_REGISTERS = [A0, A1, A2, A3, A4, A5, A6, A7, PC]
+
+    # all of the 32 bit registers
+    FULL_SIZE_REGISTERS = [D0, D1, D2, D3, D4, D5, D6, D7, A0, A1, A2, A3, A4, A5, A6, A7, PC]
```
Comment (LinuxKid21) [inline src/easier68k/core/enum/register.py]: You include A0-A7 and PC here and based off of it's usage it seems this should just be called "ALL_REGISTERS" or similar.

File: src/easier68k/simulator/m68k.py [hunk 3]
```diff
@@ -3,34 +3,150 @@
...
+        Set the registers to their default values
+        :return:
+        """
+
+        # loop through all of the full size registers which are just 32 bits / 4 bytes long
+        for register in Register.FULL_SIZE_REGISTERS:
+            self.registers[register] = bytearray(4)
+
+        # set up all of the odd registers (in this case, just the Condition Code Register)
+        # which just uses 5 bits out of the lowermost byte (do we want to allocate it an entire word instead?)
```
Comment (LinuxKid21) [inline src/easier68k/simulator/m68k.py]: I don't know, but I think that's not important for now.

File: src/easier68k/core/enum/register.py (hunk 2 above)
Comment (Chris-Johnston) [reply src/easier68k/core/enum/register.py]: I exclude the Condition Code Register

Comment (LinuxKid21) [reply src/easier68k/core/enum/register.py]: It cut out the second part of my comment...
I was meaning to say that the "MEMORY_LIMITED_ADDRESS_REGISTERS" is included in the FULL_SIZE_REGISTERS which seems like a contradiction, but I can't think of a better name anyway so I think it's fine.

Comment (Chris-Johnston) [reply src/easier68k/core/enum/register.py]: Maybe the naming could be better, the "Memory Limited" registers are 32 bits, but we restrict the values to 2^24, while the full size registers are all of the 32 bit registers (including data which are not restricted)

Comment (LinuxKid21) [reply src/easier68k/core/enum/register.py]: yeah. Let's just merge if for now.

**Pull request merged**